pkg/services/ngalert/notifier/receiver_groups.go

```diff
@@ -0,0 +1,227 @@
+package notifier
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"slices"
+
+	"github.com/grafana/alerting/notify"
+
+	"github.com/prometheus/alertmanager/config"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/auth/identity"
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/secrets"
+)
+
+var (
+	// TODO: convert to errutil
+	ErrPermissionDenied = errors.New("permission denied")
+)
+
+type ReceiverGroupService struct {
```

Comment:
Can you explain why you use "Group" in the name? Receiver in the AM terminology, which we try to adopt and migrate to, already describes a group of integrations.

Reply:
Currently we have the following external/internal names:

- `Contact Point` = `ApiReceiver`
- `Integration` = `GrafanaReceiver` =~ `EmbeddedContactPoint`
- (For testing receivers) `no UI term` = `Receiver`

I arbitrarily chose `ReceiverGroup` to avoid conflating with the `Receiver` name/model, but I'm not particularly married to this. WDYT?

Reply:
We are looking into a bright future when we do not have so many terms that describe the same conception. Last year we decided that we would adopt Alertmanager terminology. Currently, `Receiver` is the imported type. I hope in the future we will get rid of it and will reduce the scope to the generation of Alertmanager config.

So, let's stick with term Receiver even though that for now it has some ambiguous meaning.